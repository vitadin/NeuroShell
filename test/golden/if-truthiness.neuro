%% Comprehensive truthiness testing for \if command
%% Tests all possible truthy and falsy values according to isTruthy logic
\set[_echo_command="true"]

%% Explicit boolean values (case insensitive)
\if[condition=true] \echo "true is truthy"
\if[condition=TRUE] \echo "TRUE is truthy" 
\if[condition=True] \echo "True is truthy"
\if[condition=false] \echo "This should not appear - false"
\if[condition=FALSE] \echo "This should not appear - FALSE"
\if[condition=False] \echo "This should not appear - False"

%% Numeric values
\if[condition=1] \echo "1 is truthy"
\if[condition=0] \echo "This should not appear - 0"

%% Affirmative values
\if[condition=yes] \echo "yes is truthy"
\if[condition=YES] \echo "YES is truthy"
\if[condition=Yes] \echo "Yes is truthy"
\if[condition=on] \echo "on is truthy"
\if[condition=ON] \echo "ON is truthy"
\if[condition=enabled] \echo "enabled is truthy"
\if[condition=ENABLED] \echo "ENABLED is truthy"

%% Negative values
\if[condition=no] \echo "This should not appear - no"
\if[condition=NO] \echo "This should not appear - NO"
\if[condition=off] \echo "This should not appear - off"
\if[condition=OFF] \echo "This should not appear - OFF"
\if[condition=disabled] \echo "This should not appear - disabled"
\if[condition=DISABLED] \echo "This should not appear - DISABLED"

%% Empty and whitespace conditions
\if[condition=""] \echo "This should not appear - empty string"
\if[condition="   "] \echo "This should not appear - whitespace"
\if[condition="	"] \echo "This should not appear - tab"

%% Any other non-empty string should be truthy
\if[condition=hello] \echo "hello is truthy"
\if[condition=world] \echo "world is truthy"
\if[condition=random_text] \echo "random_text is truthy"
\if[condition=123abc] \echo "123abc is truthy"
\if[condition="complex string"] \echo "complex string is truthy"

%% Mixed case and edge cases
\if[condition=tRuE] \echo "This should not appear - tRuE (not exactly true)"
\if[condition=YeS] \echo "This should not appear - YeS (not exactly yes)"
\if[condition=True123] \echo "True123 is truthy (not exactly true)"

%% Special characters
\if[condition="!@#$"] \echo "!@#$ is truthy"
\if[condition="-1"] \echo "-1 is truthy"

%% Unicode characters  
\if[condition="ðŸš€"] \echo "ðŸš€ is truthy"
\if[condition="cafÃ©"] \echo "cafÃ© is truthy"

%% Test with leading/trailing whitespace (should be trimmed)
\if[condition="  true  "] \echo "Whitespace around true is trimmed"
\if[condition="  false  "] \echo "This should not appear - whitespace around false"
\if[condition="  "] \echo "This should not appear - only whitespace"

%% Verify #if_result is set for each evaluation
\if[condition=true] \set[dummy="test"]
\get[#if_result]
\if[condition=false] \set[dummy="test"]
\get[#if_result]
\if[condition=maybe] \set[dummy="test"]
\get[#if_result]