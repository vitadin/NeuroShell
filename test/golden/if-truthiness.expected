Setting _echo_command = true
%%> "\\if[condition=true] \\echo \"true is truthy\""
%%> "\\echo \"true is truthy\""
"true is truthy"
%%> "\\if[condition=TRUE] \\echo \"TRUE is truthy\""
%%> "\\echo \"TRUE is truthy\""
"TRUE is truthy"
%%> "\\if[condition=True] \\echo \"True is truthy\""
%%> "\\echo \"True is truthy\""
"True is truthy"
%%> "\\if[condition=false] \\echo \"This should not appear - false\""
%%> "\\if[condition=FALSE] \\echo \"This should not appear - FALSE\""
%%> "\\if[condition=False] \\echo \"This should not appear - False\""
%%> "\\if[condition=1] \\echo \"1 is truthy\""
%%> "\\echo \"1 is truthy\""
"1 is truthy"
%%> "\\if[condition=0] \\echo \"This should not appear - 0\""
%%> "\\if[condition=yes] \\echo \"yes is truthy\""
%%> "\\echo \"yes is truthy\""
"yes is truthy"
%%> "\\if[condition=YES] \\echo \"YES is truthy\""
%%> "\\echo \"YES is truthy\""
"YES is truthy"
%%> "\\if[condition=Yes] \\echo \"Yes is truthy\""
%%> "\\echo \"Yes is truthy\""
"Yes is truthy"
%%> "\\if[condition=on] \\echo \"on is truthy\""
%%> "\\echo \"on is truthy\""
"on is truthy"
%%> "\\if[condition=ON] \\echo \"ON is truthy\""
%%> "\\echo \"ON is truthy\""
"ON is truthy"
%%> "\\if[condition=enabled] \\echo \"enabled is truthy\""
%%> "\\echo \"enabled is truthy\""
"enabled is truthy"
%%> "\\if[condition=ENABLED] \\echo \"ENABLED is truthy\""
%%> "\\echo \"ENABLED is truthy\""
"ENABLED is truthy"
%%> "\\if[condition=no] \\echo \"This should not appear - no\""
%%> "\\if[condition=NO] \\echo \"This should not appear - NO\""
%%> "\\if[condition=off] \\echo \"This should not appear - off\""
%%> "\\if[condition=OFF] \\echo \"This should not appear - OFF\""
%%> "\\if[condition=disabled] \\echo \"This should not appear - disabled\""
%%> "\\if[condition=DISABLED] \\echo \"This should not appear - DISABLED\""
%%> "\\if[condition=\"\"] \\echo \"This should not appear - empty string\""
%%> "\\if[condition=\"   \"] \\echo \"This should not appear - whitespace\""
%%> "\\if[condition=\"\t\"] \\echo \"This should not appear - tab\""
%%> "\\if[condition=hello] \\echo \"hello is truthy\""
%%> "\\echo \"hello is truthy\""
"hello is truthy"
%%> "\\if[condition=world] \\echo \"world is truthy\""
%%> "\\echo \"world is truthy\""
"world is truthy"
%%> "\\if[condition=random_text] \\echo \"random_text is truthy\""
%%> "\\echo \"random_text is truthy\""
"random_text is truthy"
%%> "\\if[condition=123abc] \\echo \"123abc is truthy\""
%%> "\\echo \"123abc is truthy\""
"123abc is truthy"
%%> "\\if[condition=\"complex string\"] \\echo \"complex string is truthy\""
%%> "\\echo \"complex string is truthy\""
"complex string is truthy"
%%> "\\if[condition=tRuE] \\echo \"This should not appear - tRuE (not exactly true)\""
%%> "\\echo \"This should not appear - tRuE (not exactly true)\""
"This should not appear - tRuE (not exactly true)"
%%> "\\if[condition=YeS] \\echo \"This should not appear - YeS (not exactly yes)\""
%%> "\\echo \"This should not appear - YeS (not exactly yes)\""
"This should not appear - YeS (not exactly yes)"
%%> "\\if[condition=True123] \\echo \"True123 is truthy (not exactly true)\""
%%> "\\echo \"True123 is truthy (not exactly true)\""
"True123 is truthy (not exactly true)"
%%> "\\if[condition=\"!@#$\"] \\echo \"!@#$ is truthy\""
%%> "\\echo \"!@#$ is truthy\""
"!@#$ is truthy"
%%> "\\if[condition=\"-1\"] \\echo \"-1 is truthy\""
%%> "\\echo \"-1 is truthy\""
"-1 is truthy"
%%> "\\if[condition=\"ğŸš€\"] \\echo \"ğŸš€ is truthy\""
%%> "\\echo \"ğŸš€ is truthy\""
"ğŸš€ is truthy"
%%> "\\if[condition=\"cafÃ©\"] \\echo \"cafÃ© is truthy\""
%%> "\\echo \"cafÃ© is truthy\""
"cafÃ© is truthy"
%%> "\\if[condition=\"  true  \"] \\echo \"Whitespace around true is trimmed\""
%%> "\\echo \"Whitespace around true is trimmed\""
"Whitespace around true is trimmed"
%%> "\\if[condition=\"  false  \"] \\echo \"This should not appear - whitespace around false\""
%%> "\\if[condition=\"  \"] \\echo \"This should not appear - only whitespace\""
%%> "\\if[condition=true] \\set[dummy=\"test\"]"
%%> "\\set[dummy=\"test\"]"
Setting dummy = test
%%> "\\get[#if_result]"
#if_result = true
%%> "\\if[condition=false] \\set[dummy=\"test\"]"
%%> "\\get[#if_result]"
#if_result = false
%%> "\\if[condition=maybe] \\set[dummy=\"test\"]"
%%> "\\set[dummy=\"test\"]"
Setting dummy = test
%%> "\\get[#if_result]"
#if_result = true